app/assets/javascripts/service_types/IsServiceWorking.js

```diff
@@ -0,0 +1,186 @@
+import _ from 'lodash';
+import React from 'react';
+import GenericLoader from '../components/GenericLoader';
+import ExperimentalBanner from '../components/ExperimentalBanner';
+import SectionHeading from '../components/SectionHeading';
+import Button from '../components/Button';
+import {apiFetchJson} from '../helpers/apiFetchJson';
+import ProfileBarChart from '../student_profile/ProfileBarChart.js';
+import * as Routes from '../helpers/Routes';
+import moment from 'moment';
+
+const style = {
+  container: {
+    margin: 30
+  },
+  studentRow: {
+    marginBottom: 30,
+  },
+  profileBarChartStyles: {
+    title: {
+      fontSize: 20
+    },
+    container: {
+      width: '100%',
+      margin: '10px auto',
+      position: 'relative'
+    },
+    chartHeight: 250,
+  },
+  a: {
+    fontSize: 24
+  },
+  buttonStyle: {
+    float: 'left',
+    margin: 5,
+  },
+  buttonSectionStyle: {
+    width: '100%',
+    height: 50
+  }
+};
+
+class IsServiceWorking extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      serviceTypeSelected: null // options: null, 502, 503
+    };
+
+    this.fetchIsServiceWorkingData = this.fetchIsServiceWorkingData.bind(this);
+    this.renderPage = this.renderPage.bind(this);
+  }
+
+  fetchIsServiceWorkingData() {
+    const {serviceTypeSelected} = this.state;
+
+    const url = `/api/is_service_working_json/${serviceTypeSelected.id}`;
+
+    return apiFetchJson(url);
+  }
+
+  render() {
+    return (
+      <div className="IsServiceWorking">
+        <ExperimentalBanner />
+        <SectionHeading>
+          "Is this Service Working for this Student?"
+        </SectionHeading>
+        <p style={{padding: '10px 0 0 10px'}}>
+          Every student is different.
+          One student might respond well to a service, while another might respond poorly.
+        </p>
+        <p style={{padding: '10px 0 10px 10px'}}>
+          Use the buttons below to explore how students may have responded to
+          services assigned to them over the past school year.
+        </p>
+        {this.renderButtons()}
+        {this.renderCharts()}
+      </div>
+    );
+  }
+
+  renderButtons() {
+    const attendanceOfficer = {
+      id: 502,
+      name: 'Attendance Officer',
+    };
+
+    const attendanceContract = {
+      id: 503,
+      name: 'Attendance Contract',
+    };
+
+    return (
+      <div style={style.buttonSectionStyle}>
+        <Button style={style.buttonStyle}
+          onClick={() => {
+            this.setState({serviceTypeSelected: null}, () => {
+              this.setState({serviceTypeSelected: attendanceOfficer});
+            });
```

Comment:
What's with the nesting, why set it to one value and then another?  Is this here to show a loading state?

Reply:
Yes, to show loading state.

Reply:
What happens if you just call `this.setState({serviceTypeSelected: attendanceOfficer})`?  Perhaps this is something that isn't working the way it should in `GenericLoader`.

As a workaround, you could also set `key={serviceTypeSelected}` on `GenericLoader`, and that would make a new one each time that value changed, which I think would enable removing the nested `setState` calls.

Reply:
When I just call `this.setState` without the callback, it switches from one set of charts to the other without showing a "Loading..." state.

Reply:
Isn't that good? :)